pyproject.toml

```diff
@@ -54,6 +54,7 @@ dependencies = [
     "dvc-task==0.1.11",
     "dvclive>=1.2.2",
     "dvc-data==0.35.1",
+    "sqltrie",
```

Comment:
Hm, but it is a dependency of dvc-data https://github.com/iterative/dvc-data/blob/main/setup.cfg#L35 Not sure I understand why we need to explicitly declare it here.

Reply:
Seems like we should add sqltrie to https://github.com/Homebrew/homebrew-core/blob/master/pypi_formula_mappings.json , because it can't detect it there?

Reply:
ideally, we should add any extra package, as we just using whatever pypi resolves.  In this case, I think it should be declared in some where to pick up. If you do `pip install dvc[all]`, you would see that `sqltrie` missing as well.

Reply:
@chenrui333 That's very strange and would mean that we need to declare all dependencies of our dependencies, which doesn't sound right. I'm also `pip install dvc`-ing right now and it does install sqltrie all the time reliably. Am I missing something?

In the past when dep resolution tool that brew uses for python packages (sorry, don't remember the name right now) was not able to detect some dependency - we were using `pypi_formula_mapping` and that was doing the trick.

Reply:
did you do that in the `virtualenv` setup?

Reply:
Let's firstly solve the pypi dependency issue.

Reply:
You are right, I can actually see `sqltrie` in `pip install dvc`, but not in `pip install dvc[all]`.

Reply:
what is the difference between the two?

Reply:
I think it might be just a timing issue when I filed a PR. It seems working fine now.

Reply:
Unfortunately I cannot reproduce it now 😞

Reply:
So this is the flow that we generally use:
- `pip install dvc[all]`
- update deps

To answer your question earlier, the `pypi_formula_mapping` would be used for these cases:
- formula name does not match with pypi artifact
- exclude dependencies
- workaround dependency issue (which is extra_dependency)

The first two should be the primary use case, while the last one is just like workaround, kind of like the build patches that you saw in the other formulae files.

As a matter of fact, I did submit patch to fix the dependency into the upstream recently as well, like this one, https://github.com/nicolargo/glances/pull/2245

Reply:
I wish all can document those :D

Reply:
@chenrui333 https://github.com/nicolargo/glances/blob/b60f704ceb1ce087e2426545912097d3574e5c95/glances/globals.py import ujson directly, so in glances case it was a missing direct dependency, so your patch there makes sense. In dvc's case sqltrie is a dep of our dep.

The difference between `dvc` and `dvc[all]` is more extra packages for different cloud support (e.g. s3, gs, azure, etc), but the base differences (e.g. dvc-data that has sqltrie as a dep) are absolutely the same.

It seems like it is a bug on the dep resolution side of the tool that brew uses ( again, the name of that tool escapes me right now 🙁 ) and the workaround should be to include sqltrie in pypi_formula_mapping.

Thanks for your PR and for keeping an eye on the dvc formula 🙏

Reply:
Seems good on [this PR](https://github.com/Homebrew/homebrew-core/pull/121284), nothing complaining anymore.

Reply:
Not related to this PR, but this is [a PR](https://github.com/Homebrew/homebrew-core/pull/121430) for the discussions about `pypi_formula_mapping` in here.

Glad that is resolved now. 😄